### PR TITLE
MOTS-114: Locations sorting on the backend side

### PR DESCRIPTION
**Pull request by kcissewski**

### CODE REVIEW

File: src/webapp/js/components/health-workers-form.js [hunk 1]
```diff
@@ -236,7 +236,7 @@ HealthWorkersForm.propTypes = {
   onSubmitCancel: PropTypes.func.isRequired,
   onSubmit: PropTypes.func.isRequired,
   fetchLocations: PropTypes.func.isRequired,
-  availableLocations: PropTypes.shape({}),
+  availableLocations: PropTypes.arrayOf(PropTypes.shape()),
```
Comment (pmuchowski) [inline src/webapp/js/components/health-workers-form.js]: change it to PropTypes.arrayOf(PropTypes.shape({}))

File: src/webapp/js/utils/form-utils.js [hunk 2]
```diff
@@ -21,6 +21,38 @@ export function getAttributesForSelectWithClearOnChange(input, formName, ...fiel
...
 }
 
-export function sortValuesByName(object) {
-  return _.sortBy(_.values(object), x => x.name.toLowerCase());
+export function getChiefdomsFromDistrictById(list, id) {
+  const chiefdoms = [];
+  if (list && id) {
+    list.forEach((loc) => {
+      if (loc.id === id) {
+        Array.prototype.push.apply(chiefdoms, loc.chiefdoms);
```
Comment (pmuchowski) [inline src/webapp/js/utils/form-utils.js]: why you don't just return the loc.chiefdoms here?

File: src/webapp/js/utils/form-utils.js [hunk 3]
```diff
@@ -21,6 +21,38 @@ export function getAttributesForSelectWithClearOnChange(input, formName, ...fiel
   };
 }
 
-export function sortValuesByName(object) {
-  return _.sortBy(_.values(object), x => x.name.toLowerCase());
+export function getChiefdomsFromDistrictById(list, id) {
```
Comment (pmuchowski) [inline src/webapp/js/utils/form-utils.js]: I think you can change the name "list" to "districts"

**Comment by pmuchowski** [comment]
You will also need to change the locations in the mobile app

**Comment by awalkowiak** [comment]
Would not something like this: https://github.com/motech-implementations/mots/blob/develop/mobile/utils/form-utils.js#L8 work here too (for getting available locs in each select field)?

**Comment by pmuchowski** [comment]
this needs to be changed, because locations are not stored in maps anymore

**Comment by awalkowiak** [comment]
@pmuchowski yeah, i am aware of this. However I suggested this solution to eliminate a few separate functions for extracting specified location type. Beside i wanted to show current solution that matched one of your review comment. (cc @kcissewski )

**Comment by kcissewski** [comment]
@awalkowiak Thank you for suggesting better way to implementing it. I used this solution on mobile side already and I think that it also can be used on web side. Thanks!

File: mobile/utils/form-utils.js [hunk 4]
```diff
@@ -5,19 +5,30 @@ import styles from '../styles/inputsStyles';
 
 const { labelSelectFieldStyle, optionListStyle } = styles;
 
-export function getSelectableLocations(locations, districtId, chiefdomId, facilityId) {
+function getLocationById(list, id) {
+  if (list && id) {
+    for (let i = 0; i < list.length; i += 1) {
```
Comment (awalkowiak) [inline mobile/utils/form-utils.js]: I think you could use _.filter here.

File: src/webapp/js/utils/form-utils.js [hunk 5]
```diff
@@ -21,6 +21,35 @@ export function getAttributesForSelectWithClearOnChange(input, formName, ...fiel
   };
 }
 
-export function sortValuesByName(object) {
-  return _.sortBy(_.values(object), x => x.name.toLowerCase());
+export function getChiefdomsFromDistrictById(districts, districtId) {
+  if (districts && districtId) {
+    for (let i = 0; i < districts.length; i += 1) {
```
Comment (awalkowiak) [inline src/webapp/js/utils/form-utils.js]: _.filter should work here too

File: src/webapp/js/utils/form-utils.js [hunk 6]
```diff
@@ -21,6 +21,35 @@ export function getAttributesForSelectWithClearOnChange(input, formName, ...fiel
...
+        return districts[i].chiefdoms;
+      }
+    }
+  }
+  return [];
+}
+
+export function getFacilitiesFromChiefdomById(chiefdoms, chiefdomId) {
+  if (chiefdoms && chiefdomId) {
+    for (let i = 0; i < chiefdoms.length; i += 1) {
```
Comment (awalkowiak) [inline src/webapp/js/utils/form-utils.js]: _.filter should work here too

File: src/webapp/js/utils/form-utils.js [hunk 7]
```diff
@@ -21,6 +21,35 @@ export function getAttributesForSelectWithClearOnChange(input, formName, ...fiel
...
+        return chiefdoms[i].facilities;
+      }
+    }
+  }
+  return [];
+}
+
+export function getCommunitiesFromFacilitiesById(facilities, facilityId) {
+  if (facilities && facilityId) {
+    for (let i = 0; i < facilities.length; i += 1) {
```
Comment (awalkowiak) [inline src/webapp/js/utils/form-utils.js]: _.filter should work here too

**Pull request merged**